onert-micro/eval-driver/TrainingDriver.cpp

```diff
@@ -100,118 +92,180 @@ int entry(int argc, char **argv)
-  luci_interpreter::Interpreter interpreter(model_data.data(), true);
-
-  luci_interpreter::training::TrainingSettings settings;
-  settings.learning_rate = 0.0001;
-  settings.number_of_epochs = 100;
-  settings.batch_size = 1;
-
-  const auto input_size = interpreter.getInputDataSizeByIndex(0);
-  const auto output_size = interpreter.getOutputDataSizeByIndex(0);
-
-  char *train_data = new char[input_size * num_train_data_samples];
-  char *label_train_data = new char[output_size * num_train_data_samples];
-
-  char *test_data = new char[input_size * num_test_data_samples];
-  char *label_test_data = new char[output_size * num_test_data_samples];
-
-  readDataFromFile(input_train_data_path, train_data, input_size * num_train_data_samples);
-  readDataFromFile(input_test_data_path, test_data, input_size * num_test_data_samples);
-
-  readDataFromFile(input_label_train_data_path, label_train_data,
-                   output_size * num_train_data_samples);
-  readDataFromFile(input_label_test_data_path, label_test_data,
-                   output_size * num_test_data_samples);
+  return model_data;
+}
 
-  luci_interpreter::training::TrainingOnertMicro onert_micro_training(&interpreter, settings);
-  onert_micro_training.enableTrainingMode();
-  onert_micro_training.train(num_train_data_samples, reinterpret_cast<uint8_t *>(train_data),
-                             reinterpret_cast<uint8_t *>(label_train_data));
-  onert_micro_training.disableTrainingMode();
+} // namespace
 
-  auto size = interpreter.getInputDataSizeByIndex(0);
-  auto train_data_u8 = reinterpret_cast<char *>(train_data);
+/*
+ * @brief EvalDriver main
+ *
+ *  Driver for testing training onert micro
+ *  Current example for testing classification task
+ *
+ */
+int entry(int argc, char **argv)
+{
+  if (argc != 10)
+  {
+    std::cerr << "Usage: " << argv[0]
+              << " <path/to/circle/model/without/weights> "
+                 " <path/to/wof/file> <path/to/save/wof/file> <path/to/input/train_data> "
+                 "<path/to/input/target_train_data> "
+                 "<path/to/input/test_data> <path/to/input/target_test_data> num_of_train_smpl "
+                 "num_of_test_smpl\n";
+    return EXIT_FAILURE;
+  }
 
-  printf("Run train dataset:\n");
-  for (int i = 0; i < num_train_data_samples; ++i)
+  const char *circle_model_path = argv[1];
+  const char *wof_file_path = argv[2];
+  const char *output_wof_file_path = argv[3];
+  const char *input_input_train_data_path = argv[4];
+  const char *input_target_train_data_path = argv[5];
+  const char *input_input_test_data_path = argv[6];
+  const char *input_target_test_data_path = argv[7];
+  const int32_t num_train_data_samples = atoi(argv[8]);
+  const int32_t num_test_data_samples = atoi(argv[9]);
+
+  DataBuffer circle_model = readFile(circle_model_path);
+  DataBuffer wof_data = readFile(wof_file_path);
+
+  // Configure training mode
+  onert_micro::OMConfig config;
+  config.wof_ptr = wof_data.data();
```

Comment:
```suggestion
  config.wof_ptr = nullptr;
```
For june release, let's train circle-with-weight format.

Reply:
Then we need to figure out how to save the training result

Reply:
Introduced  #13107 (without weight divider), please check

Reply:
> Then we need to figure out how to save the training result

As you already implement at next PR, final weight should be saved as single circle file. Intermediate checkpoint keeps weight and other ongoing state at another file.